src/clib/pio_darray.c

```diff
@@ -1484,11 +1484,11 @@ static int check_adios2_need_to_flush(file_desc_t *file, adios_var_desc_t *av)
 }
 
 static void *PIOc_convert_buffer_adios(file_desc_t *file, io_desc_t *iodesc,
-                                       adios_var_desc_t *av, void *array, int arraylen,
+                                       adios_var_desc_t *av, const void *array, int arraylen,
                                        int *ierr)
 {
     assert(file != NULL && iodesc != NULL && av != NULL && array != NULL && ierr != NULL);
-    void *buf = array;
+    void *buf = NULL;
```

Comment:
If the ADIOS_CONVERT* macros use buf it should be passed in as an arg to the macro

Reply:
OK, I will apply this minor change to  ADIOS_CONVERT* macros.